components/server/src/api/configuration-service-api.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { HandlerContext, ServiceImpl } from "@connectrpc/connect";
+import { inject, injectable } from "inversify";
+import { ConfigurationService as ConfigurationServiceInterface } from "@gitpod/public-api/lib/gitpod/experimental/v2/configuration_connect";
+import {
+    CreateConfigurationRequest,
+    CreateConfigurationResponse,
+    DeleteConfigurationRequest,
+    DeleteConfigurationResponse,
+    GetConfigurationRequest,
+    GetConfigurationResponse,
+    ListConfigurationsRequest,
+    ListConfigurationsResponse,
+} from "@gitpod/public-api/lib/gitpod/experimental/v2/configuration_pb";
+import { PublicAPIConverter } from "@gitpod/gitpod-protocol/lib/public-api-converter";
+import { PaginationResponse } from "@gitpod/public-api/lib/gitpod/experimental/v2/pagination_pb";
+import { ProjectsService } from "../projects/projects-service";
+
+@injectable()
+export class ConfigurationServiceAPI implements ServiceImpl<typeof ConfigurationServiceInterface> {
+    constructor(
+        @inject(ProjectsService)
+        private readonly projectService: ProjectsService,
+        @inject(PublicAPIConverter)
+        private readonly apiConverter: PublicAPIConverter,
+    ) {}
+
+    async createConfiguration(
+        req: CreateConfigurationRequest,
+        context: HandlerContext,
+    ): Promise<CreateConfigurationResponse> {
+        if (!req.organizationId) {
+            throw new Error("organizationId is required");
```

Comment:
we should use gRPC errors, i.e. `new ConnectError("organizationId is required", Code.InvalidArgument)` or os

Otherwise it will be treated as internal error.

Reply:
Should we not use ApplicationError here?

Reply:
Yeah, it is going to be important for the dashboard for now.